src/code/z_map_data.c

```diff
@@ -1,79 +1,577 @@
+    switch (MapData_MID_GetType(mapId)) {
+        case MAPDATA_MID_GAMEPLAY_DANGEON_KEEP:
+            MapData_GetMapTexDimGameplayDangeonKeep(mapId, width, height);
+            return;
+        case MAPDATA_MID_MAP_GRAND_STATIC:
+            MapData_GetMapTexGameplayDangeonKeepDim(mapId, width, height);
+            return;
+        default:
+            *width = *height = 0;
+            return;
+    }
+}
+
+void MapData_GetMapTexOffset(s32 mapId, s32* offsetX, s32* offsetY) {
+    switch (MapData_MID_GetType(mapId)) {
+        case MAPDATA_MID_GAMEPLAY_DANGEON_KEEP:
+            MapData_GetMapTexOffsetGameplayDangeonKeep(mapId, offsetX, offsetY);
+            return;
+        case MAPDATA_MID_MAP_GRAND_STATIC:
+            MapData_GetMapTexGameplayDangeonKeepOffset(mapId, offsetX, offsetY);
+            return;
+        default:
+            *offsetX = *offsetY = 0;
+            return;
+    }
+}
+
+void MapData_GetMapScale(s32 mapId, s32* scale) {
+    switch (MapData_MID_GetType(mapId)) {
+        default:
+            *scale = 0;
+            return;
+        case MAPDATA_MID_GAMEPLAY_DANGEON_KEEP:
+            *scale = MapData_GetMapGameplayDangeonKeepScale(mapId);
+            return;
+        case MAPDATA_MID_MAP_GRAND_STATIC:
+            *scale = MapData_GetMapGrandScale(mapId);
+            return;
+    }
+}
+
+void MapData_GetDrawType(s32 mapId, s32* drawType) {
+    switch (MapData_MID_GetType(mapId)) {
+        case MAPDATA_MID_GAMEPLAY_DANGEON_KEEP:
+            MapData_GetMapDrawTypeGameplayDangeonKeep(mapId, drawType);
+            return;
+        case MAPDATA_MID_MAP_GRAND_STATIC:
+            MapData_GetMapGrandDrawType(mapId, drawType);
+            return;
+        default:
+            *drawType = 0;
+            return;
+    }
+}
+
+s32 MapData_GetMapColorIndex(s32 mapId) {
+    switch (MapData_MID_GetType(mapId)) {
+        case MAPDATA_MID_GAMEPLAY_DANGEON_KEEP:
+            return MapData_GetMapColorIndexGameplayDangeonKeep(mapId);
+        case MAPDATA_MID_MAP_GRAND_STATIC:
+            return MapData_GetMapGrandColorIndex(mapId);
+        default:
+            return 0;
+    }
+}
+
+s32 func_80109964(s32 arg0) {
+    return D_801BF15C[arg0];
+}
+
+void MapData_GetMapITexDim(s32 mapId, s32* width, s32* height) {
+    *width = map_i_static_Info[mapId].texWidth;
+    *height = map_i_static_Info[mapId].texHeight;
+}
+
+void MapData_GetMapITexOffset(s32 mapId, s32* offsetX, s32* offsetY) {
+    *offsetX = map_i_static_Info[mapId].offsetX;
+    *offsetY = map_i_static_Info[mapId].offsetY;
+}
+
+void MapData_GetMapIDrawType(s32 mapId, s32* drawType) {
+    *drawType = map_i_static_Info[mapId].drawType;
+}
+
+u8 MapData_GetMapIColorIndex(s32 mapId) {
+    return map_i_static_Info[mapId].colorIndex;
+}
+
+s16 MapData_GetMapIScale(s32 mapId) {
+    return map_i_static_Info[mapId].scale;
+}
+
+s32 MapData_MapITexSizeTest(s32 mapId) {
+    s32 i;
+    s32 size;
+
+    size = 0;
```

Comment:
```suggestion
    s32 size = 0;

```
This kind of comment could apply to a few functions, could you double check these?